### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by SimbaWANGU**
In this Pull Request, the following changes have been introduced:
1. Add a column species of type string to your animals table. Modify your schema.sql file and populated it with the following data:
- Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
- Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
- Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
- Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
- Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
- Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
- Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

and added these insertions to your data.sql file.

2. Updated the species column in a transaction:
i) To unspecified and rollback
 
unspecified

ii) To digimon and pokemon
species transaction

3. In another transaction, all records were deleted:

delete transaction

and rolledback:

rollback delete transaction

4. In a seperate transaction, a number of queries were run to do the following:
- Deleted all animals born after Jan 1st, 2022 and Created a savepoint for the transaction.

transaction 3 upto savepoint

- Update all animals' weight to be their weight multiplied by -1.

multiply weight by -1

- Rollback to the savepoint

rollback to savepoint

- Update all animals' weights that are negative to be their weight multiplied by -1 and Commit transaction

update negative weights

5. Wrote queries to answer the following questions
- How many animals are there?

animal count

- How many animals have never tried to escape?

never tried to escape

- What is the average weight of animals?

average

- Who escapes the most, neutered or not neutered animals?

escapes

- What is the minimum and maximum weight of each type of animal?

min and max weight by species

- What is the average number of escape attempts per animal type of those born between 1990 and 2000?

average escape attempts

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -7,5 +7,6 @@ CREATE TABLE animals(
 	escape_attempts INT NOT NULL,
 	neutered BOOLEAN NOT NULL,
 	weight_kg DECIMAL NOT NULL,
-    PRIMARY KEY(id)
+	species VARCHAR NOT NULL,
```
Comment (SajjadAhmad14) [inline schema.sql]: Please use `ALTER TABLE` command to modify the `animals` table.

File: queries.sql [hunk 2]
```diff
@@ -22,4 +22,47 @@ SELECT * FROM animals WHERE neutered = 't';
 SELECT * FROM animals WHERE name NOT LIKE '%Gabumon%';
 
 /* Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg) */
-SELECT * FROM animals WHERE weight_kg >= '10.4' AND weight_kg <= '17.3';
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg >= '10.4' AND weight_kg <= '17.3';
+
```
Comment (SajjadAhmad14) [inline queries.sql:26]: Please update the `animals` table by setting the species column to `unspecified`, then roll back the change and verify that change was made.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Simba, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
README is professional. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by SimbaWANGU** [comment]
@SajjadAhmad14 thank you for the review.
Changing the species column to 'unspecified' does not require a screenshot as per the requirements.
I have included it in my next commit nonetheless.

File: queries.sql [hunk 3]
```diff
@@ -22,4 +22,47 @@ SELECT * FROM animals WHERE neutered = 't';
...
+/* Update animals table column species in transaction and cpmmit */
```
Comment (BenMukebo) [inline queries.sql]: - [x] As requested by the previous reviewer [here](https://github.com/SimbaWANGU/vetclinic_db/pull/2/files#r914031612) 🙃 Kindly make sure to address this 👇🏻 
- Inside a transaction update the `animals` table by setting the `species` column to `unspecified`. Verify that change was made. Then roll back the change and verify that the `species` columns went back to the state before the transaction.

**Review by BenMukebo — CHANGES_REQUESTED** [review]
Hi @SimbaWANGU,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight :clap: 
- GitHub flow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️


 ## Required Changes ♻️
_Check the comments under the review._
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please Tag me on (@BenMukebo) when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @SimbaWANGU , 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: Well implemented suggested changes
:heavy_check_mark: Descriptive readme file.

### Optional suggestions :arrow_down_small: 

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**